### DISCUSSION TITLE
Theming the way dark mode works?

### DISCUSSION THREAD

**Discussion by henriklangerhansson** [tailwindlabs/tailwindcss#12531] [Help] [answered]
I'm exploring an approach to implement basic theming. In some projects, I've used dark mode using `darkMode: 'class'` in `tailwind.config.ts`. This simplifies tasks like, for example, altering the background color of the header on specific pages by adding `class="dark"` to the layout template. However, this approach has limitations as it eliminates the option to use dark mode or multiple themes. I'm sure there's an elegant solution that I have overlooked!

**Comment by alex-krasikau** [comment, marked as answer]
Hello @henrikhansson!

Here are a few solutions:

1) When you switch to class mode for dark mode, it gives more options. Many companies do this by finding the default theme from user choices with `window.matchMedia('(prefers-color-scheme: dark)')`, and then they save it in localstorage. You can also put a theme switch on top. There is a next.js library that does this for you - [https://github.com/pacocoursey/next-themes](https://github.com/pacocoursey/next-themes).

2) About multitheming, you can add new themes easily using the `addVariant` plugin API. Here is how you can do it:
   ```js
   plugins: [
       require("tailwindcss/plugin")(function ({addVariant}) {
           addVariant("blue", `:is(.blue &)`)
       }),
   ],
   ```

3) I suggest using custom variables as your main method. Only use `dark:` and `blue:` patterns in special cases. This makes your code easier and better to keep up. For example, look here - [https://tailwindcss.com/docs/customizing-colors#using-css-variables](https://tailwindcss.com/docs/customizing-colors#using-css-variables)

**Reply by henriklangerhansson** [reply]
Thank you for your quick and thorough response!

**Reply by q1b** [reply]
You can check below link, @henrikhansson

https://www.protailwind.com/workshops/multi-theme-strategy
